File: artexploration/src/main/java/com/example/artexploration/part1/TestActivity.java
package com.example.artexploration.part1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.pm.PackageManager;
import android.content.pm.ResolveInfo;
import android.content.res.Configuration;
import android.net.Uri;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import com.example.artexploration.R;


public class TestActivity extends AppCompatActivity {

    private static final String TAG = "tag";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_test);
        Log.d(TAG, "onCreate");
        Button button = findViewById(R.id.bt_first);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //隐式启动
                Intent intent = new Intent();
                intent.setAction("com.example.artexploration.part1.action");
                intent.addCategory("com.category.test2");
                intent.setDataAndType(Uri.parse("content://abc"),"image/png");
                startActivity(intent);

            }
        });
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putString("bundle","test");
        Log.d(TAG, "onSaveInstanceState:");
    }

    @Override
    protected void onRestoreInstanceState(Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
        String test = savedInstanceState.getString("bundle");
        Log.d(TAG, "restore:"+test);
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d(TAG, "onStart");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d(TAG, "onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d(TAG, "onPause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d(TAG, "onStop");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.d(TAG, "onRestart");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d(TAG, "onDestroy");
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        Log.d(TAG, "onConfigurationChanged");
    }
}

File: developers/src/main/java/com/example/developers/api/TestActivity.kt
package com.example.developers.api

import android.animation.Animator
import android.animation.AnimatorInflater
import android.animation.AnimatorSet
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import androidx.databinding.DataBindingUtil
import com.example.developers.R
import com.example.developers.databinding.ActivityTestBinding

class TestActivity : AppCompatActivity() {

    private lateinit var dataBinding:ActivityTestBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        dataBinding= DataBindingUtil.setContentView(this,R.layout.activity_test)
        activity1()
    }

    private fun animation() {
        val hyperspaceJump: Animation = AnimationUtils.loadAnimation(this,R.anim.hyperspace_jump)
        dataBinding.view.startAnimation(hyperspaceJump)

    }

    private fun propertyAnimator() {
        val set: Animator? = AnimatorInflater.loadAnimator(this,R.animator.property_animator)
                .apply {
                    setTarget(dataBinding.view)
                    start()
                }

    }

    private fun activity1(){
        val intent = Intent(Intent.ACTION_SEND).apply {
            putExtra(Intent.EXTRA_EMAIL,"dst")
        }
        startActivity(intent)
    }
}
